pandas/tests/test_expressions.py

```diff
@@ -431,3 +432,30 @@ def test_bool_ops_column_name_dtype(self, test_input, expected):
         # GH 22383 - .ne fails if columns containing column name 'dtype'
         result = test_input.loc[:, ["a", "dtype"]].ne(test_input.loc[:, ["a", "dtype"]])
         assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "axis,arith", product((0, 1), ("add", "sub", "mul", "mod", "truediv"))
+    )
+    def test_frame_series_axis(self, axis, arith):
+        # Can't check floordiv here because it currently doesn't work #GH27636
+        df = self.frame
+        if axis == 1:
+            other = self.frame.iloc[0, :]
+        else:
+            other = self.frame.iloc[:, 0]
+
+        expr._MIN_ELEMENTS = 0
+
+        op = lambda x, y: getattr(x, arith)(y, axis=axis)
+        op.__name__ = arith
+
+        expr.set_use_numexpr(False)
+        expected = op(df, other)
+        expr.set_use_numexpr(True)
+
+        result = op(df, other)
+        try:
+            assert_frame_equal(expected, result)
+        except Exception:
```

Comment:
why do you need to catch and re-raise?

Reply:
Was following the convention in run_arithmetic. Happy to change to just the assert if that is preferable

Reply:
Yea I think just assert